### PR TITLE
NAS-113034 / 22.02-RC.2 / prevent some db queries on passive

### PR DESCRIPTION
**Pull request by yocalebo**
We do not need to add these database operations to the `JournalSync` thread's queue to synchronize to the standby controller.

### CODE REVIEW

File: src/middlewared/middlewared/plugins/migration.py [hunk 1]
```diff
@@ -50,6 +50,6 @@ async def run(self):
                     self.middleware.logger.error("Error running migration %s", name, exc_info=True)
                     continue
 
-                await self.middleware.call("datastore.insert", "system.migration", {"name": name})
+                await self.middleware.call("datastore.insert", "system.migration", {"name": name}, {"ha_sync": False})
 
             await self.middleware.call("keyvalue.set", "run_migration", False)
```
Comment (themylogin) [inline src/middlewared/middlewared/plugins/migration.py:55]: Also need to change it here

File: src/middlewared/middlewared/plugins/failover_/journal.py [hunk 2]
```diff
@@ -118,7 +118,8 @@ def _handle_sql_queue_item(self, item):
                 self.journal.append(item)
             else:
                 query, params = item
-                logger.warning('Node status %s but executed SQL query: %s', self.failover_status, query)
+                if query and query.find('services_ssh') == -1:
```
Comment (themylogin) [inline src/middlewared/middlewared/plugins/failover_/journal.py:121]: Can't you just use `ha_sync: false` for the update host keys query?

**Review by themylogin — CHANGES_REQUESTED** [review]

File: src/middlewared/middlewared/plugins/failover_/journal.py (hunk 2 above)
Comment (yocalebo) [reply src/middlewared/middlewared/plugins/failover_/journal.py]: I've reverted that change and will add it to my open PR here: https://github.com/truenas/middleware/pull/7843

File: src/middlewared/middlewared/plugins/keyvalue.py [hunk 3]
```diff
@@ -37,22 +37,28 @@ async def get(self, key, default):
...
         except IndexError:
-            await self.middleware.call("datastore.insert", "system.keyvalue", {
-                "key": key,
-                "value": json.dumps(value)
-            })
+            await self.middleware.call(
+                "datastore.insert", "system.keyvalue", {"key": key, "value": json.dumps(value)}, options
+            )
         else:
             await self.middleware.call("datastore.update", "system.keyvalue", row["id"], {
```
Comment (themylogin) [inline src/middlewared/middlewared/plugins/keyvalue.py:53]: Also need to pass options here

**Review by themylogin — CHANGES_REQUESTED** [review]

File: src/middlewared/middlewared/plugins/keyvalue.py (hunk 3 above)
Comment (yocalebo) [reply src/middlewared/middlewared/plugins/keyvalue.py]: oops, thanks

**Review by themylogin — APPROVED** [review]

**Pull request merged**